src/sentry/remote_subscriptions/consumers/result_consumer.py

```diff
@@ -54,7 +60,40 @@ def handle_result(self, subscription: U | None, result: T):
 
 
 class ResultsStrategyFactory(ProcessingStrategyFactory[KafkaPayload], Generic[T, U]):
-    def __init__(self) -> None:
+    parallel_executor: ThreadPoolExecutor | None = None
+
+    parallel = False
+    """
+    Does the consumer process unrelated messages in parallel?
+    """
+
+    max_batch_size = 500
+    """
+    How many messages will be batched at once when in parallel mode.
+    """
+
+    max_batch_time = 10
+    """
+    The maximum time in seconds to accumulate a bach of check-ins.
+    """
+
+    def __init__(
+        self,
+        mode: Literal["parallel", "serial"] = "serial",
+        max_batch_size: int | None = None,
+        max_batch_time: int | None = None,
+        max_workers: int | None = None,
+    ) -> None:
```

Comment:
Do we need to setup the options in the `sentry.consumers` module like this as well https://github.com/getsentry/sentry/blob/e58f8b971c8bee0beb7eee22eef457d71125ec50/src/sentry/consumers/__init__.py#L107-L135 ?